force-app/main/default/classes/GetProcessInstanceDataTest.cls

```diff
@@ -1,95 +1,151 @@
 @isTest
 public with sharing class GetProcessInstanceDataTest {
 
-   
-   
-    public static void setup() {
-        //create a record
-        // Insert an account
-        Account curAccount = new Account(Name='Test',annualRevenue=100.0);
-        insert curAccount;
+    @TestSetup
+    private static void makeData(){
+        //insert a record that has an approval process on the targeted object
+        Account account = new Account();
+            account.Name = 'Test Account';
+        insert account;
+    }
 
-        User testUser1 = insertStandardUser();
+    @isTest
+    private static void getProcessInstanceDataMainTest() {
+        //query that record
+        Account sob = [
+            SELECT Id, 
+                Name 
+            FROM Account 
+            LIMIT 1
+        ];
+
+        String sObjName = sob.getSObjectType().getDescribe().getName();
+
+        //query the approval process for the record
+        ProcessDefinition processDefinition = [
+            SELECT Id, 
+                Name, 
+                Description, 
+                DeveloperName, 
+                LockType, 
+                State, 
+                Type,
+                TableEnumOrId
+            FROM ProcessDefinition
+            WHERE State = 'Active' 
+                AND Type = 'Approval'
+                AND TableEnumOrId =: sObjName
+            LIMIT 1 
+        ];
+        
+        //submit the record into the approval
+        Approval.ProcessSubmitRequest request = new Approval.ProcessSubmitRequest();
+            request.setComments('Submitting request for approval.');
+            request.setObjectId(sob.id);
+            request.setSubmitterId(UserInfo.getUserId()); 
+            request.setProcessDefinitionNameOrId(processDefinition.DeveloperName);
+            request.setSkipEntryCriteria(true);
+        Approval.ProcessResult result = Approval.process(request);
+
+
+        //query the user / queue that the record will be assigned to once submitted for approval - this will be your ActorId
+        Group queue = [
```

Comment:
Query the Queue or User that is assigned the first step in the approval process. For Example - The 'Processor_Assignor' queue gets assigned the first approval step in the process.

![image](https://user-images.githubusercontent.com/36901822/199027037-1ad888dc-7272-44f3-9c5b-d3031d54dcdf.png)